src/herder/TxSetFrame.cpp

```diff
@@ -851,15 +1076,19 @@ TxSetFrame::addTxsFromXdr(Application& app,
 #ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
         tx->maybeComputeSorobanResourceFee(ledgerVersion, sorobanConfig,
                                            appConfig);
+        // Phase should be consistent with the tx we're trying to add
+        if (tx->isSoroban() && phase != Phase::SOROBAN)
```

Comment:
What if the opposite condition is true? Should this be just `if (tx->isSoroban() != Phase::SOROBAN)`?

Reply:
good catch. i'd prefer to be explicit here though, so changed to `((tx->isSoroban() && phase != Phase::SOROBAN) || (!tx->isSoroban() &&  phase != Phase::CLASSIC))`